Data Structures/Linked-Lists/Circular Linked List/deletion.cpp

```diff
@@ -0,0 +1,52 @@
+/* Deletes a node of given address in a circular linked list where last node's next points to first node.
+   There are 3 cases: 1. node to be deleted is the last node
+                      2. node to be deleted is the only node in list.
+                      3. node to be deleted is present at the beginning of list
+   
+   To run this code, make sure you make appropriate calls to this function in main().
+*/
+
+
+struct node {
+ 
+  int data;
+  node *next;
+};
+
+/* Deletes a node of given address
+    node** - address of the HEAD (beginning of linked list)
+    given_node* - node to be deleted(its address is passed as argument)
+*/
+void del(node **head, node *given_node){
```

Comment:
proper function name "delete"

Reply:
"delete" is a keyword in c++, it will case an error if I name it delete.

Reply:
Okay, then why not deleteNode?
I mean to say that it should have a proper name.

Reply:
Sure! I'll do that.